src/snakemake/sourcecache.py

```
@@ -222,28 +315,75 @@
+        checkout = self._checkout()
+        return checkout.mtime(self.path)
+
+    def _checkout(self):
+        try:
+            return self.hosted_repo.checkout(self.branch, self.commit or self.tag)
+        except Exception as e:
+            raise WorkflowError(
+                "Failed to clone/checkout git repository "
+                f"{self.host}/{self.repo} at {self.branch or self.commit or self.tag}.",
+                e,
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)